### ISSUE TITLE
question regarding addons

### ISSUE THREAD

**Issue by iamandymcinnes** [polinchw/cluster-addons#2]
Hi!

First of all, thanks for the detailed writeup on autopilot and argo. I just had a question around the addons helm charts... What is the reasoning behind having the helm chart files in your repo, as opposed to just using the helm chart package and then using a kustomization to specify the version and your values file?

And secondly, does using the application set allow argo to track the changes for each of the applications (addons) where using an application directly wouldn't? I notice changing the application set itself doesn't trigger a sync without touching the application directory in the autopilot repo for the cluster addons, where as changes to the addons does.

Thanks again

Andy

**Comment by polinchw** [comment]
Andy,

Thanks for reaching out to me about the argocd cluster addons.

I'll do my best to answer your questions:

1. Q. What is the reasoning behind having the helm chart files in the repo as apposed to having the helm chart package?
A: I feel I have more control over what gets deployed in the helm chart if it is untarred in the repo.  For example if I wanted to add something to the helm template I could just add it to the repo and argocd would automatically pick up the change.

2. Q. Does the application set allow argocd to track changes for each application?
A. Yes.  If I were to introduce a new application or change an existing application, for example change a values-dev.yaml setting for grafana, argocd would automatically pick up the change.  ArgoCD is looking at this repo in its entirety.  Argocd sees each app folder as a separate ArgoCD Application.  

I have a video that shows what happens when I commit changes to the cluster-addons repo that a live kubernetes cluster is pointing to.

[YouTube video](https://youtu.be/K48CHU9MUJI?t=2228)

**Comment by iamandymcinnes** [comment]
Hi 

Thanks for your answer, I've replicated your cluster addons and also have it syncing, as you describe if I change things within the addons directory. However if I change something in the base it did not sync. I found the only way to get argo to sync that type of change, was to update a ref in the overlay path for the cluster addon app in the autopilot repo. As in by adding a ref to this line or simply touching a file in the apps directory in the autopilot repo: https://github.com/polinchw/auto-pilot/blob/43221c135eea5b5d6bc9e1f133503f8881966546/apps/addons/base/kustomization.yaml#L4

Many thanks

Andy
